starlite/storage/file_backend.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import shutil
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from anyio import Lock, Path
+from anyio.to_thread import run_sync
+
+from .base import StorageBackend, StorageObject
+
+if TYPE_CHECKING:
+    from os import PathLike
+
+
+class FileStorageBackend(StorageBackend):
+    """Session backend to store data in files."""
+
+    __slots__ = ("path", "_lock")
+
+    def __init__(self, path: PathLike) -> None:
+        self._lock = Lock()
+        self.path = Path(path)
+
+    @staticmethod
+    async def _load_from_path(path: Path) -> StorageObject:
+        data = await path.read_bytes()
+        return StorageObject.from_bytes(data)
+
+    async def get(self, key: str, renew: int | None = None) -> bytes | None:
+        path = self.path / key
+        async with self._lock:
+            if not await path.exists():
```

Comment:
Checking for a paths existence before doing anything with it is considered a race condition. Best to catch FileNotFoundError in _load_from_path

Reply:
Good point